src/main/java/stsjorbsmod/actions/DamageWithOnKillEffectAction.java

```diff
@@ -0,0 +1,43 @@
+package stsjorbsmod.actions;
+
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.cards.DamageInfo;
+import com.megacrit.cardcrawl.core.AbstractCreature;
+import com.megacrit.cardcrawl.powers.MinionPower;
+
+public class DamageWithOnKillEffectAction extends AbstractGameAction {
+    private DamageInfo info;
+    private Runnable onKillCallback;
+    private boolean worksOnMinions;
+
+
+    public DamageWithOnKillEffectAction(AbstractCreature target, DamageInfo info, Runnable onKillCallback) {
+        this.info = info;
```

Comment:
You can use a delegating constructor here instead of repeating the block of settings (ie, this first ctor's body can be just `this(target, info, onKillCallback, false);`